tests/testcafe/tests.js

```diff
@@ -50,6 +50,35 @@ function prepareClient() {
   document.querySelector('base').parentNode.removeChild(document.querySelector('base'));
 }
 
+function hiddenTest(game, variant, md5, tests) {
+  const testUrl = `${server}/library/${game}.vtt#VTT`;
+  test.after(async t => {
+    await removeGame(t);
+    await t.expect(Selector('#statesOverlay').visible).ok();
+  })(`Public library: ${game} (variant ${variant})`, async t => {
+    await ClientFunction(prepareClient)();
+    await t
+      .pressKey('esc')
+      .click('#statesButton')
+      .setNativeDialogHandler(() => testUrl, { dependencies: { testUrl }})
+      .click('button.prettyButton.link:nth-child(2)')
+      .pressKey('enter')
+      .hover('.roomState')
+      .click(Selector('button.play').nth(variant));
+    await setName(t);
+    await tests(t);
+    await compareState(t, md5);
+  });
+}
+
+function hiddenLibraryButtons(game, variant, md5, buttons) {
+  hiddenTest(game, variant, md5, async t => {
+    for(const b of buttons) {
+      await t.click(`[id="${b}"]`).wait(4000);
+    }
+  });
+}
+
```

Comment:
This is mostly just copied code. That's fine for testing but you should try to merge it with the other functions when it is working.

Reply:
Agreed.